sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1454,6 +1547,9 @@ pi_result _pi_queue::executeCommandList(pi_command_list_ptr_t CommandList,
       // command-list itself. This host-visible event will not be
       // waited/released by SYCL RT, so it must be destroyed after all events in
       // the batch are gone.
+      // We know that refcount is more than 2 so calling piEventRelease is safe
+      // because it locks the queue only if refcount of the event is zero.
+      assert(HostVisibleEvent->RefCount > 2);
```

Comment:
Why can't it be exactly 2?

Reply:
It can't be exactly 2 because I check that EventList of the command list is not empty above, i.e. after createEventAndAssociateQueue ref count is 2 and then +1 for each event in the EventList. This code makes sense only if EventList is not empty, more over before this change it was possible to get a situation when deleted Event is dereferenced if EventList is empty.

Reply:
It's not just an optimization. Problem is that if even list is empty then HostVisible event will be deleted after the second call to piEventRelease and we use it afterwards https://github.com/intel/llvm/pull/6179/files#diff-15dd1eb076d2164bd9e87d9057f05f652a716498e8cdf5975e564c65309a0985R1465-R1473.

Reply:
I see. Please add a comment about this where you check for non-empty list